src/ui/components/SuggestedResults/index.js

```diff
@@ -0,0 +1,54 @@
+/* @flow */
+/* eslint react/no-array-index-key: 0 */
+import React from 'react';
+import escapeStringRegexp from 'escape-string-regexp';
+
+import './style.scss';
+
+type Props = {
+  query: string,
+  suggestions: Array<string>,
+  onSelect: Function,
+};
+
+const renderSuggestion = (query: string, suggestion: string, onSelect: Function) => {
+  let parts;
+  if (query.replace(/\s+/, '') === '') {
+    parts = [suggestion];
+  } else {
+    parts = suggestion.split(new RegExp(`(${escapeStringRegexp(query)})`, 'gi'));
+  }
+
+  const onClick = (e) => {
+    e.preventDefault();
+    onSelect(suggestion);
+  };
+
+  return (
+    <li key={suggestion}>
+      <button className="SuggestedResults-item" onClick={onClick}>
+        {parts.map((text, idx) => {
+          if (text.toLowerCase() === query.toLowerCase()) {
+            return (
+              <span className="is-highlighted" key={`${suggestion}-${text}=${idx}`}>
+                {text}
+              </span>
+            );
+          }
+
+          return text;
+        })}
+      </button>
+    </li>
+  );
+};
+
+const SuggestedResults = ({ query, suggestions, onSelect }: Props) => (
+  <div className="SuggestedResults">
+    <ul className="SuggestedResults-list">
+      {suggestions.map((s) => renderSuggestion(query, s, onSelect))}
```

Comment:
Just as a separate point (for future reference), I prefer to see any function with more than one argument to take an object of parameters, like `renderSuggestion({ query, suggestion: s, onSelect }))`. It makes it easier to look at the function call and see what's going on.